nomic/tests/test_atlas_client.py

```diff
@@ -382,3 +382,56 @@ def test_map_embeddings():
     assert 'my_tag' not in map.tags.get_tags()
 
     project.delete()
+
+
+def test_map_text_pandas():
+    size = 50
+    data = pd.DataFrame({
+        'field': [str(uuid.uuid4()) for i in range(size)],
+        'id': [str(uuid.uuid4()) for i in range(size)],
+        'color': [random.choice(['red', 'blue', 'green']) for i in range(size)],
+    })
+
+    project = atlas.map_text(
+        name='UNITTEST_pandas_text',
+        id_field='id',
+        indexed_field="color",
+        data=data,
+        is_public=True,
+        colorable_fields=['color'],
+        reset_project_if_exists=True,
+    )
+
+    map = project.get_map(name='UNITTEST_pandas_text')
+
+    assert project.total_datums == 50
+
+    project.delete()
+
+
+def test_map_text_iterator():
+    size = 50
+    data = [
+        {
+            'field': str(uuid.uuid4()),
+            'id': str(uuid.uuid4()),
+            'color': random.choice(['red', 'blue', 'green'])
+        } 
+        for _ in range(size)
+    ]
+
+    data_iter = iter(data)
+
+    project = atlas.map_text(
+        name='UNITTEST_pandas_text',
+        id_field='id',
+        indexed_field="color",
+        data=data_iter,
+        is_public=True,
+        colorable_fields=['color'],
+        reset_project_if_exists=True,
+    )
+
+    map = project.get_map(name='UNITTEST_pandas_text')
```

Comment:
Should not re-use the project name here for both tests with `reset_project_if_exists`; seems like pytest doesn't run in parallel by default, but maybe we'll use one of the packages to speed up the tests some day.